### PR TITLE
fix norns view highlighting and separate it from note highlights

### PR DESCRIPTION
**Pull request by entzmingerc**
adds another array for tracking norns keyboard layout highlighting separate from incoming notes to highlight on other keyboard layouts

### CODE REVIEW

File: src/deluge/model/instrument/melodic_instrument.cpp [hunk 1]
```diff
@@ -297,9 +287,22 @@ void MelodicInstrument::offerReceivedNote(ModelStackWithTimelineCounter* modelSt
                                           int32_t midiChannel, int32_t note, int32_t velocity, bool shouldRecordNotes,
                                           bool* doingMidiThru) {
 	MIDIMatchType match = midiInput.checkMatch(fromDevice, midiChannel);
+	InstrumentClip* instrumentClip = (InstrumentClip*)activeClip;
```
Comment (sapphire-arches) [inline src/deluge/model/instrument/melodic_instrument.cpp:290]:
```suggestion
	auto *instrumentClip = static_cast<InstrumentClip*>(activeClip);
```

File: src/deluge/gui/ui/keyboard/layout/norns.cpp [hunk 2]
```diff
@@ -47,9 +47,9 @@ void KeyboardLayoutNorns::renderPads(RGB image[][kDisplayWidth + kSideBarWidth])
 		for (int32_t x = 0; x < kDisplayWidth; x++) {
 			int32_t note = noteFromCoords(x, y);
 
-			// If highlighting notes is active, do it
-			if (getHighlightedNotes()[note] != 0) {
-				image[y][x] = colours::white_full.adjust(getHighlightedNotes()[note], 1);
+			// If norns is active, do it
+			if (getNornsNotes()[note] != 0) {
+				image[y][x] = colours::white_full.adjust(getNornsNotes()[note], 1);
```
Comment (sapphire-arches) [inline src/deluge/gui/ui/keyboard/layout/norns.cpp:52]: does this break regular note highlighting?

Comment (entzmingerc) [reply src/deluge/gui/ui/keyboard/layout/norns.cpp:52]: It shouldn't. Previously, the incoming midi notes from midigrid intended for updating the pad LEDs in norns view would be saved to the highlightedNotes array. If you opened a clip with another keyboard layout, it'd render all the noise from midigrid. Or if you had incoming midi notes, it'd interfere with the norns view and highlight things that shouldn't be lit up. I tested it a bit last night where I had an incoming midi sequence on channel 1 highlighting on isomorphic and the norns view highlighting from midigrid stream.

**Comment by n3ptun3s** [comment]
Is this fix in the latest nightly?
I updated today. The led feedback is working - norns sends stuff to the deluge,
but the deluge does not send midi to the norns :(

**Comment by entzmingerc** [comment]
> Is this fix in the latest nightly? I updated today. The led feedback is working - norns sends stuff to the deluge, but the deluge does not send midi to the norns :(

nope, just this PR currently, it'll get there eventually

**Review by m-m-adams — COMMENTED** [review]
Changes look to be addressed, lgtm

**Pull request merged**